cache/directory.go

```diff
@@ -0,0 +1,44 @@
+package cache
+
+import (
+	"context"
+	"errors"
+
+	"google.golang.org/protobuf/types/known/anypb"
+	"google.golang.org/protobuf/types/known/emptypb"
+
+	"github.com/pomerium/pomerium/pkg/grpc/databroker"
+	"github.com/pomerium/pomerium/pkg/grpc/directory"
+)
+
+// RefreshUser refreshes a user's directory information.
+func (c *Cache) RefreshUser(ctx context.Context, req *directory.RefreshUserRequest) (*emptypb.Empty, error) {
+	c.mu.Lock()
```

Comment:
Sorry, I don't fully understand the mutex here. Can you explain?

Reply:
When config is updated it will reset the directory provider: https://github.com/pomerium/pomerium/pull/1460/files/047a53f8a71ec724941ebaf9ecce91a529232823#diff-f7b51df04a56b62e71e037c6cacb4d6bR141

The mutex is necessary to avoid a race on reading/writing the variable.

Reply:
Got it. Thanks